src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSAAndroid.cs

```diff
@@ -0,0 +1,839 @@
+            {
+                // In the event that a key was loaded via ImportParameters or an IntPtr/SafeHandle
+                // it could be outside of the bounds that we currently represent as "legal key sizes".
+                // Since that is our view into the underlying component it can be detached from the
+                // component's understanding.  If it said it has opened a key, and this is the size, trust it.
+                KeySizeValue = newKeySize;
+            }
+
+            public override KeySizes[] LegalKeySizes
+            {
+                get
+                {
+                    return new[] { new KeySizes(512, 16384, 8) };
+                }
+            }
+
+            public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
+            {
+                if (data == null)
+                    throw new ArgumentNullException(nameof(data));
+                if (padding == null)
+                    throw new ArgumentNullException(nameof(padding));
+
+                Interop.AndroidCrypto.RsaPadding rsaPadding = GetInteropPadding(padding, out RsaPaddingProcessor? oaepProcessor);
+                SafeRsaHandle key = GetKey();
+
+                int rsaSize = Interop.AndroidCrypto.RsaSize(key);
+                byte[]? buf = null;
+                Span<byte> destination = default;
+
+                try
+                {
+                    buf = CryptoPool.Rent(rsaSize);
+                    destination = new Span<byte>(buf, 0, rsaSize);
+
+                    if (!TryDecrypt(key, data, destination, rsaPadding, oaepProcessor, out int bytesWritten))
+                    {
+                        Debug.Fail($"{nameof(TryDecrypt)} should not return false for RSA_size buffer");
+                        throw new CryptographicException();
+                    }
+
+                    return destination.Slice(0, bytesWritten).ToArray();
+                }
+                finally
+                {
+                    CryptographicOperations.ZeroMemory(destination);
+                    CryptoPool.Return(buf!, clearSize: 0);
+                }
+            }
+
+            public override bool TryDecrypt(
+                ReadOnlySpan<byte> data,
+                Span<byte> destination,
+                RSAEncryptionPadding padding,
+                out int bytesWritten)
+            {
+                if (padding == null)
+                {
+                    throw new ArgumentNullException(nameof(padding));
+                }
+
+                Interop.AndroidCrypto.RsaPadding rsaPadding = GetInteropPadding(padding, out RsaPaddingProcessor? oaepProcessor);
+                SafeRsaHandle key = GetKey();
+
+                int keySizeBytes = Interop.AndroidCrypto.RsaSize(key);
+
+                // OpenSSL does not take a length value for the destination, so it can write out of bounds.
+                // To prevent the OOB write, decrypt into a temporary buffer.
```

Comment:
Oh, and a pass to remove OpenSSL-specific comments is probably also good :smile: